File: restful-server/src/main/java/via/sep/restful_server/notification/dto/PropertyNotificationDTO.java
package via.sep.restful_server.notification.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.util.Map;

@Data
@AllArgsConstructor
public class PropertyNotificationDTO {
    private String propertyId;
    private String address;
    private BigDecimal price;
    private String propertyType;
    private Map<String, Object> details;
}

File: Blazor/RealEstateBlazor/Data/DTOs/PropertyNotificationDTO.cs
using RealEstateBlazor.Services;

namespace RealEstateBlazor.Data.DTOs;

public class PropertyNotificationDTO : INotification
{
    public string PropertyId { get; set; }
    public string Address { get; set; }
    public decimal Price { get; set; }
    public string PropertyType { get; set; }
    public Dictionary<string, object> Details { get; set; }
    public DateTime Timestamp { get; set; }
}
